pkg/crds/crds.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crds
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"time"
+)
+
+var cephResources = []string{
+	"cephclusters",
+	"cephblockpoolradosnamespaces",
+	"cephblockpools",
+	"cephbucketnotifications",
+	"cephbuckettopics",
+	"cephclients",
+	"cephcosidrivers",
+	"cephfilesystemmirrors",
+	"cephfilesystems",
+	"cephfilesystemsubvolumegroup",
+	"cephnfses",
+	"cephobjectrealms",
+	"cephobjectstores",
+	"cephobjectstoreusers",
+	"cephobjectzonegroups",
+	"cephobjectzones",
+	"cephrbdmirrors",
+}
+
+const (
+	cephRookIoGroup          = "ceph.rook.io"
+	cephRookResourcesVersion = "v1"
+)
+
+const (
+	CephResourceKindCephCluster = "CephCluster"
+	toolBoxDeployment           = "rook-ceph-tools"
+	destroyProcessTimeOut       = 2 * time.Minute
+)
+
+var (
+	clusterResourcePatchFinalizer = map[string]interface{}{
+		"spec": map[string]interface{}{
+			"cleanupPolicy": map[string]string{
+				"confirmation": "yes-really-destroy-data",
+			},
+		},
+	}
+
+	defaultResourceRemoveFinalizers = map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"finalizers": nil,
+		},
+	}
+)
+
+func DeleteCustomResources(ctx context.Context, clientsets k8sutil.ClientsetsInterface, k8sClientSet kubernetes.Interface, clusterNamespace string) {
+	err := deleteCustomResources(ctx, clientsets, clusterNamespace)
+	if err != nil {
+		logging.Fatal(err)
+	}
+	k8sutil.DeleteDeployment(ctx, k8sClientSet, clusterNamespace, toolBoxDeployment)
+	time.Sleep(destroyProcessTimeOut)
```

Comment:
Instead of a sleep, let's specifically wait for the cleanup job to exist, and to be completed. Two minutes might be enough, but it might not be. We don't want to print "done" until the jobs are all completed. Perhaps this means we loop for a while and every 15s check if it exists and if it's completed, and if not print a statement that we're waiting. Then the user knows we are waiting for the job(s) to complete.